cmd/kueuectl/app/version/version_test.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	cmdtesting "sigs.k8s.io/kueue/cmd/kueuectl/app/testing"
+)
+
+func TestVersionCmd(t *testing.T) {
+	testCases := map[string]struct {
+		deployment *v1.Deployment
+		args       []string
+		wantOut    string
+		wantOutErr string
+		wantErr    error
+	}{
+		"should print client version": {
+			args:    []string{},
+			wantOut: "Client Version: v0.0.0-main\n",
+		},
+		"should print client and server versions": {
+			deployment: &v1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      kueueControllerManagerName,
+					Namespace: kueueNamespace,
+				},
+				Spec: v1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "manager",
+									Image: "registry.k8s.io/kueue/kueue:v0.0.0",
```

Comment:
NIT: Should we verify the version that it follows semVer?
Maybe some negative test, like wrongly formatted version.

Reply:
Like this 0.0.0-rc.1?

Reply:
Or you mean `latest`?

Reply:
In my opinion it better to print it even if not semver.